lib/routes/asianfanfics/tag.ts

```diff
@@ -0,0 +1,112 @@
+import { DataItem, Route } from '@/types';
+import logger from '@/utils/logger';
+import { parseDate } from '@/utils/parse-date';
+import puppeteer from '@/utils/puppeteer';
+import { load } from 'cheerio';
+
+// test url http://localhost:1200/asianfanfics/tag/milklove/N
+
+export const route: Route = {
+    path: '/tag/:tag/:type',
+    categories: ['reading'],
+    example: '/tag/milklove/N',
+    parameters: {
+        tag: '标签',
+        type: '排序类型',
+    },
+    name: '亚洲同人网标签',
+    maintainers: ['KazooTTT'],
+    radar: [
+        {
+            source: ['www.asianfanfics.com/browse/tag/:tag/:type'],
+            target: '/tag/:tag/:type',
+        },
+    ],
+    description: `匹配亚洲同人网标签，支持类型：
+- L: Latest 最近更新
+- N: Newest 最近发布
+- O: Oldest 最早发布
+- C: Completed 已完成
+- OS: One Shots 短篇
+`,
+    handler,
+    features: {
+        requirePuppeteer: true,
+    },
+};
+
+type Type = 'L' | 'N' | 'O' | 'C' | 'OS';
+
+const typeToText = {
+    L: '最近更新',
+    N: '最近发布',
+    O: '最早发布',
+    C: '已完成',
+    OS: '短篇',
+};
+
+async function handler(ctx) {
+    const tag = ctx.req.param('tag');
+    let type = ctx.req.param('type') as Type;
+    if (!type || !['L', 'N', 'O', 'C', 'OS'].includes(type)) {
+        type = 'L';
+    }
```

Comment:
https://github.com/DIYgod/RSSHub/blob/322e0d5401f555b039b1ad829cc2ca494c1408c6/lib/routes/asianfanfics/tag.ts#L10

If `type` is a compulsory parameter, no default value should be given